app/scripts/lib/hardware-keyring-builder-factory.ts

```diff
@@ -0,0 +1,30 @@
+import { KeyringClass, Json } from '@metamask/utils';
+
+/**
+ * A transport bridge between the keyring and the hardware device.
+ */
+export type HardwareTransportBridge = {
+  new (): HardwareTransportBridge;
```

Comment:
A new HardwareTransportBridge returns a type of itself? It feels like this type ought to be named something else 🤔

Reply:
Sorry, I realize this comment may have been confusing. It feels like we need one type to represent the constructor (and any static properties that exist on it) and another to represent the instance. Agree/disagree?

Reply:
I do agree, this exported type is indeed wrong. I just changed this to:

```typescript
export type HardwareTransportBridgeClass = new () => TrezorBridge;
```

When implementing the Ledger bridge, perhaps we can simply change this to:

```typescript
export type HardwareTransportBridgeClass = 
    new () => TrezorBridge | LedgerBridge;
```